### DISCUSSION TITLE
Sources for REI Plugin Compatibility mod

### DISCUSSION THREAD

**Discussion by SquidDev** [shedaniel/RoughlyEnoughItems#1154] [Q & A]
I'm currently in the process of porting some JEI compatibility code to use REI's API. For this I was hoping to consult how the REI Plugin Compatibilities mapped between the two.

However, I noticed that the source for this isn't actually available on trunk, and the distributed jar was obfuscated. I was a bit confused why this was, especially as [the sources are on other branches](https://github.com/shedaniel/RoughlyEnoughItems/tree/d4db5d6d9a042d5f4d85b2eaafb1cda649a171fd). Would it be possible to publish the sources for on trunk too, purely as a reference for people more familiar with JEI's API?

**Comment by MaxNeedsSnacks** [comment]
While I‘m personally in favour of this, I unfortunately also understand the reason and backstory behind why REI plugin compatibilities was obfuscated and made private; without getting into too much detail, and because this is shedaniel‘s story to tell, someone was basically constantly asking him to port the layer to fabric so they could *stop* using REI‘s API entirely and solely rely on what the compat layer does for them for compatibility (shoveling off all support burden to hin mind you), and after not getting their way they decided to „threaten“ to port it themselves, which ultimately led to this current mess. Again I don’t think this was a good choice, but it should be noted the compat layer is not really meant for devs since it relies… pretty heavily on internals iirc, and is really mostly meant to be a user facing tool for people who want to get into REI. I think it’s definitely worth talking to shedaniel about this again when he is less… preoccupied with other matters such as right now, but until then, the last public version of the compat layer should also still be basically up to date as a reference  ^^

**Reply by SquidDev** [reply]
Thank you for the additional context, that does sound miserable to deal with.
